nixos/modules/services/web-apps/keycloak.nix

```
@@ -637,7 +637,7 @@ in
               cp $CREDENTIALS_DIRECTORY/ssl_{cert,key} /run/keycloak/ssl/
             '' + ''
               export KEYCLOAK_ADMIN=admin
-              export KEYCLOAK_ADMIN_PASSWORD=${cfg.initialAdminPassword}
+              export KEYCLOAK_ADMIN_PASSWORD="${cfg.initialAdminPassword}"
```

I think this should maybe be

```suggestion
              export KEYCLOAK_ADMIN_PASSWORD=${lib.escapeShellArg cfg.initialAdminPassword}
```

Which should handle e.g. a password containing a `"`.